servers/xr/xr_interface.h

```diff
@@ -150,6 +153,14 @@ class XRInterface : public RefCounted {
 	virtual XRInterface::EnvironmentBlendMode get_environment_blend_mode() const { return XR_ENV_BLEND_MODE_OPAQUE; }
 	virtual bool set_environment_blend_mode(EnvironmentBlendMode mode) { return false; }
 
+	/* VRS */
+	float get_vrs_min_radius() const;
+	void set_vrs_min_radius(float p_vrs_min_radius);
+	float get_vrs_strength() const;
+	void set_vrs_strength(float p_vrs_strength);
+	virtual Vector2 get_eye_focus(uint32_t p_view, float p_aspect); /* Obtain eye focus (if supported) in 2D space NDC coordinates to given view */
+	virtual RID get_vrs_texture(); /* obtain VRS texture */
```

Comment:
These two are OK, because they are `virtual`. I'm not really fond of the default implementation of `get_vrs_texture()` (which was already there before this PR, it's just updated here) because it's very RD specific, and not all `XRInterface`'s are necessarily going to use RD. It's not the worst thing, because it's `virtual` and so can be overridden. But, still, it would have been better to have that implementation else where and call it if the `XRInterface` supports VRS and uses RD, rather than having it be a part of all `XRInterface`'s automatically.

Reply:
Actually its not. I've standardised the VRS texture to always be an RG2 texture, whether supplied by the XR system or by the user (vrs_mode = VRS_TEXTURE), then inside the renderer it's converted to what is actually needed. Turns out Vulkan fragment rate, vulkan density map, DX12 and likely Metal all have different requirements here. So this approach should be compatible with the OpenGL renderer if we ever implement our own VRS solution there.

That is also the reason I put this default logic here, this implementation will work on any headset. It's virtual because certain back ends might have their own logic.

Reply:
> That is also the reason I put this default logic here, this implementation will work on any headset.

Well, except for the interfaces where VRS won't work at all. For example, this won't work on WebXR because it only works through the compatibility renderer and this default implementation uses RD. I should probably update `WebXRInterface`'s `get_vrs_texture()` to just return a `RID()` because that function won't do anything useful there.

Reply:
Yup, when it's not supported, it's not supported. It won't even ever call get_vrs_texture in that case as it's gated on the renderer having VRS support.